.github/workflows/tests.yml

```diff
@@ -0,0 +1,27 @@
+name: tests
+on:
+  push:
+    branches:
+      - master
+      - main
+      - release-*
+  pull_request:
+  workflow_dispatch:
+jobs:
+  unit-tests:
+    strategy:
+      matrix:
+        os: [ubuntu, windows, masOS]
+    runs-on: ${{ matrix.os }}-latest
```

Comment:
It is better to use fixed os versions for prevent unexpected break unit tests when github update "latest" image version.